### DISCUSSION TITLE
Npm build fails with Nextjs local API giving error of "typeerror: fetch failed at object.fetch"

### DISCUSSION THREAD

**Discussion by InamAli123** [vercel/next.js#59088] [App Router]
I have created a basic nextjs app with new app router 13.5.6 and I am using nextjs API for backend along with Prisma. However, everything works great but it is not building with npm build and giving error of fetch failed. I am aware that this error occurs because the next server is down during build process and the app is unable to fetch data from the backend. But what is the work around of this issue. I also tried to write the backend code directly in the server component instead of sending fetch request still no luck. If someone has solved this issue, then please help. 
Thank You. 
![Capture](https://github.com/vercel/next.js/assets/46759690/97b28118-3528-469c-abb1-a76e3de919c4)

**Comment by paulrogov** [comment]
Here is a response by Next developer:
https://github.com/vercel/next.js/issues/44062#issuecomment-1653822615

In short:  
add this line at the beginning of a page that fails to be built  
`export const dynamic = 'force-dynamic'`

It works at least for now (Next 14, Node v20).

**Reply by S-Nishant** [reply]
This makes the page to opt out from SSG, in my case I am fetching the data from route api and creating SSG for the same page.

Adding `export const dynamic = 'force-dynamic'` removes the build error but the page skips out the data I was expecting to view after the build
